tests/QueryTest.php

```diff
@@ -327,6 +327,20 @@ public function testPaginate(): void
         $this->assertEquals(1, $results->currentPage());
     }
 
+    public function testThatPaginateCastsPerPageParameter()
+    {
+        $results = User::paginate('2');
```

Comment:
What is the reason pass $perPage as string, if original [method](https://github.com/laravel/framework/blob/7.x/src/Illuminate/Database/Eloquent/Builder.php#L732-L741) expect int or null ?

Reply:
See issue: #1750

Reply:
Tests in laravel don't have case with pass string `$perPage` https://github.com/laravel/framework/blob/0b12ef19623c40e22eff91a4b48cb13b3b415b25/tests/Integration/Database/QueryBuilderTest.php#L181
https://github.com/laravel/framework/blob/0b12ef19623c40e22eff91a4b48cb13b3b415b25/tests/Database/DatabaseQueryBuilderTest.php#L3205

Reply:
So... Whats the problem with adding additional tests?

Reply:
I don't understand why we  pass not int value

Reply:
Because it's possible when using the default laravel model without mongodb

Reply:
i saw in [docs](https://laravel.com/docs/7.x/pagination) and tests and didn't find examples with passing string `$perPage`

Reply:
To make you happy, i just tested it in a stock laravel installation and confirmed that it actually does work in a stock installation.
And to no surprise it does.
Granted, when i created the PR i didn't fact check the original issue